include/bedrock/world/level/block/block.h

```diff
@@ -56,9 +56,9 @@ class Block {
         return legacy_block_->requiresCorrectToolForDrops();
     }
 
-    [[nodiscard]] bool canContainLiquid() const
+    [[nodiscard]] DetectionRule getLiquidDetectionRule() const
```

Comment:
This method doesn't really exist in the Block class in BDS. Better add the `const BlockComponentDirectData &getDirectData() const;` method which returns the `direct_data_` and access `water_detection_rule` from there.